### PR TITLE
[ENH] Add `hasPortalURI` to Dataset model

### LINKED ISSUE #141: [ENH] Add portal URI and datalad IRI to dataset model
### Is there an existing issue for this?

- [X] I have searched the existing issues

### New feature

In https://github.com/neurobagel/project/issues/119 we decided that the dataset model needs to gain two new attributes:
1. Something to contain a dereferencable URI to a website describing the data for humans, and possibly access instructions (optional). Working title `hasPortalURI`
~2. Something to contain a filesystem path or datalad ID to where the data are located (ideally machine resolvable). (required). Working title `hasFilePath`~

### Steps to implement
- [X] add optional `hasPortalURI` attribute to `Dataset` model
  - [x] expand help text: option represents a website/webpage with dataset access instructions and/or a dataset description (almost like a DOI - since most datasets with a URL will include access instructions)
~- [X] add required  `hasFilePath` attribute to `Dataset` model~
- [x] Keep `--pheno` and `--dictionary` required opts for `bagel pheno`
~- [x] For `bagel pheno` command, add a required command line arg `--dataset_dir`~
~- [x] Change `--pheno` and `--dictionary` to be optional and, if provided, must be located somewhere inside the `--dataset_dir` (one level separation only)~
  - [x] (**already in README**) clear documentation how users running CLI in container can still get the full file system path "parsed" by `bagel bids` (e.g. "please bind dataset dir to same path inside container").
- [x] Probably some type of visual / logging confirmation on what path the `bagel pheno` and `bagel bids` "sees" for the dataset to help fix the problem of people not reading these great new documentation examples
- [x] Dataset `hasPortalURI` should probably be user supplied via another optional arg in `bagel pheno`
  - [x] implement + test some validation to check that it's a URL
- [x] update tests of outputted `@context` to include new dataset attributes
- [x] regenerate example_synthetic.jsonld and example_synthetic_pheno-bids.jsonld
- [x] add tests that these attributes appear correctly in the output jsonld
- [x] update README

### PR DESCRIPTION
**Pull request by alyssadai**
Closes #141 

Changes proposed in this pull request:

- New optional CLI arg `--portal` added
- Validation of `--portal` value as HttpUrl
- `hasPortalURI` added to data model
- Added tests for valid/invalid portal URLs

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[BUG]`, `[DOC]`, `[INFRA]`, `[MAINT]`)
- [x] PR links to Github issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Code is properly formatted

For new features:
- [ ] Tests have been added

### CODE REVIEW

File: bagel/cli.py [hunk 1]
```diff
@@ -58,6 +65,14 @@ def pheno(
     pheno_df = pd.read_csv(pheno, sep="\t", keep_default_na=False, dtype=str)
     putil.validate_inputs(data_dictionary, pheno_df)
 
+    # Display validated input paths to user
+    space = 25
```
Comment (alyssadai) [inline bagel/cli.py:69]: Neat!

File: bagel/tests/test_cli_pheno.py [hunk 2]
```diff
@@ -81,6 +81,48 @@ def test_invalid_inputs_are_handled_gracefully(
...
+@pytest.mark.parametrize(
+    "portal",
+    [
+        "openneuro.org/datasets/ds002080",
+        "https://openneuro",
+        "not a url",
+        "www.github.com/mycoolrepo/mycooldataset",
+    ],
+)
+def test_invalid_portal_uris_produces_stderr(
```
Comment (alyssadai) [inline bagel/tests/test_cli_pheno.py:93]: For clarity, rename test: 
```suggestion
def test_invalid_portal_uris_produces_error(
```

File: bagel/tests/test_cli_pheno.py [hunk 3]
```diff
@@ -81,6 +81,48 @@ def test_invalid_inputs_are_handled_gracefully(
...
+    runner,
+    test_data,
+    tmp_path,
+    portal,
+):
+    """Tests that invalid or non-HTTP/HTTPS URLs result in a user-friendly error in the stdout."""
```
Comment (alyssadai) [inline bagel/tests/test_cli_pheno.py:99]:
```suggestion
    """Tests that invalid or non-HTTP/HTTPS URLs result in a user-friendly error."""
```

File: bagel/tests/test_cli_pheno.py [hunk 4]
```diff
@@ -81,6 +81,48 @@ def test_invalid_inputs_are_handled_gracefully(
...
+            "--name",
+            "test dataset 2",
+            "--portal",
+            portal,
+        ],
+        catch_exceptions=False,
+    )
+
+    assert result.exit_code != 0
+    # For some reason, it seems the Rich formatting of the stdout error causes problems
```
Comment (alyssadai) [inline bagel/tests/test_cli_pheno.py:119]:
```suggestion
    # For some reason, it seems the Rich formatting of the error causes problems
```

**Review by alyssadai — COMMENTED** [review]
Looks great 🍒

**Review by rmanaem — APPROVED** [review]
🧑‍🍳

**Pull request merged**